src/parsers/perf/perfparser.cpp

```diff
@@ -1091,7 +1112,25 @@ class PerfParserPrivate : public QObject
             }
         };
 
-        bottomUpResult.addEvent(type, sampleCost.cost, sample.frames, frameCallback);
+        const auto costAggregation = Settings::instance()->costAggregation();
+
+        switch (costAggregation) {
+        case Settings::CostAggregation::BySymbol:
+            bottomUpResult.addEvent(type, sampleCost.cost, sample.frames, frameCallback);
+            break;
+        case Settings::CostAggregation::ByThread:
+            bottomUpResult.addEvent({QString::number(sample.tid)}, type, sampleCost.cost, sample.frames);
```

Comment:
instead of using the raw tid here, map it directly to a thread name via

```
commands.value(sample.pid).value(sample.tid)
```

Reply:
does this also apply to `ByProcess`? In that case I can remove the lookup in line 844